leases/app/controllers/MediaLeaseController.scala

```diff
@@ -38,13 +37,14 @@ object MediaLeaseController extends Controller
 
   val notFound = respondNotFound("MediaLease not found")
 
-  val appIndex = AppIndex(
-    "media-leases",
-    "Media leases service",
-    Map(),
-    List(Link("by-media-id", s"$rootUri/leases/media/{id}"))
-  )
-  def index = Authenticated { _ => respond(appIndex) }
+
+  val indexResponse = {
+    val appIndex = AppIndex("media-leases", "Media leases service", Map())
+    val indexLinks =  List(Link("by-media-id", s"$rootUri/leases/media/{id}"))
```

Comment:
Came across this before, vaguely considered refactoring so that URI's were
not required. Gave up. Calling @theefer to see if he has an opinion.
On Wed, 20 Apr 2016 at 22:43, Nicholas Papacostas <notifications@github.com>
wrote:

> In leases/app/controllers/MediaLeaseController.scala
> <https://github.com/guardian/grid/pull/1865#discussion_r60493231>:
>
> > @@ -38,13 +37,14 @@ object MediaLeaseController extends Controller
> >
> >    val notFound = respondNotFound("MediaLease not found")
> >
> > -  val appIndex = AppIndex(
> > -    "media-leases",
> > -    "Media leases service",
> > -    Map(),
> > -    List(Link("by-media-id", s"$rootUri/leases/media/{id}"))
> > -  )
> > -  def index = Authenticated { _ => respond(appIndex) }
> > +
> > +  val indexResponse = {
> > +    val appIndex = AppIndex("media-leases", "Media leases service", Map())
> > +    val indexLinks =  List(Link("by-media-id", s"$rootUri/leases/media/{id}"))
>
> I was thinking this really belongs in "Actions" not "Links", so I tried
> doing that, but in a surprise twist the "{ }" in the URL is incompatible
> with java.net.uri which is needed for Action
>
> The solutions I found online seemed overly clunky and made the point that
> actually "{}" and not allowed in URIs >.>
> Open to ideas 🉐
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/guardian/grid/pull/1865/files/e13d8a565019866b6430deb059466977aa568b2e#r60493231>
>

-- 

------------------------------
This e-mail and all attachments are confidential and may also be 
privileged. If you are not the named recipient, please notify the sender 
and delete the e-mail and all attachments immediately. Do not disclose the 
contents to another person. You may not use the information for any 
purpose, or store, or copy, it in any way.  Guardian News & Media Limited 
is not liable for any computer viruses or other material transmitted with 
or as part of this e-mail. You should employ virus checking software.
 
Guardian News & Media Limited is a member of Guardian Media Group plc. Registered 
Office: PO Box 68164, Kings Place, 90 York Way, London, N1P 2AP.  Registered 
in England Number 908396